### PR TITLE
[Feature][Connector V2] Add GoogleSheets Sink

### PR DESCRIPTION
**Pull request by TaoZex**
## Purpose of this pull request

improve pr : https://github.com/apache/incubator-seatunnel/pull/3411

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

**Comment by TaoZex** [comment]
I don't have permission to update this pr:https://github.com/apache/incubator-seatunnel/pull/3411
So I created this pr to improve it. @EricJoy2048

**Review by CalvinKirs — CHANGES_REQUESTED** [review]
Please provide UT or IT to verify this, you can set to disable, but we should have a test code that can be quickly and easily configured and then run.

**Comment by TaoZex** [comment]
> Please provide UT or IT to verify this, you can set to disable, but we should have a test code that can be quickly and easily configured and then run.

Thanks for your advice. I will add e2e test code.

**Comment by TaoZex** [comment]
@Hisoka-X @EricJoy2048 PTAL

**Comment by TaoZex** [comment]
@hailin0 @Hisoka-X @EricJoy2048 PTAL

File: docs/en/connector-v2/sink/GoogleSheets.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+> GoogleSheets sink connector
+## Description
+
+Used to write data to GoogleSheets.
+
+## Key features
+
+- [ ] [exactly-once](../../concept/connector-v2-features.md)
+- [ ] [schema projection](../../concept/connector-v2-features.md)
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/GoogleSheets.md]: `schema projection` is not sink key features. Please reference https://seatunnel.apache.org/docs/concept/connector-v2-features.

File: seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/sink/SheetsSinkWriter.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        BatchUpdateValuesRequest body = new BatchUpdateValuesRequest()
+                .setValueInputOption("RAW")
+                .setData(data);
+        service.spreadsheets().values().batchUpdate(sheetsParameters.getSheetId(), body).execute();
+        seaTunnelRowList.clear();
+    }
+
+    @Override
+    public void close() throws IOException {
+        // not need close
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/sink/SheetsSinkWriter.java]: Need `flush()` ?

Comment (TaoZex) [reply seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/sink/SheetsSinkWriter.java]: Thanks. I have fixed it.

**Comment by TaoZex** [comment]
@EricJoy2048 I have resolved conflict. PTAL.

File: docs/en/connector-v2/sink/GoogleSheets.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# GoogleSheets
+
+> GoogleSheets sink connector
+>
+  ## Description
```
Comment (TyrantLucifer) [inline docs/en/connector-v2/sink/GoogleSheets.md:5]: remove blank

File: seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/config/SheetsParameters.java [hunk 4]
```diff
@@ -42,4 +63,27 @@ public SheetsParameters buildWithConfig(Config config) {
...
+    public Sheets buildSheets() throws IOException {
+        byte[] keyBytes = Base64.getDecoder().decode(this.serviceAccountKey);
+        ServiceAccountCredentials sourceCredentials =
+                ServiceAccountCredentials.fromStream(new ByteArrayInputStream(keyBytes));
+        sourceCredentials =
+                (ServiceAccountCredentials)
+                        sourceCredentials.createScoped(
+                                Collections.singletonList(SheetsScopes.SPREADSHEETS));
+        HttpRequestInitializer requestInitializer = new HttpCredentialsAdapter(sourceCredentials);
+        NetHttpTransport httpTransport = null;
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/config/SheetsParameters.java:76]:
```suggestion
        NetHttpTransport httpTransport;
```

File: seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/exception/GoogleSheetsConnectorErrorCode.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        return this.code;
+    }
+
+    @Override
+    public String getDescription() {
+        return this.description;
+    }
+
+    @Override
+    public String getErrorMessage() {
```
Comment (TyrantLucifer) [inline seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/exception/GoogleSheetsConnectorErrorCode.java:45]: Remove

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]
Update release-note plugin_config seatunnel-dist

image

**Comment by TyrantLucifer** [comment]
Offser the snapshot images of this connectors in different engines (flink spark zeta), not the snapshot images of seatunnel-examples, you need compile it and run test cases in integration test env.

**Comment by hailin0** [comment]
waitting ci

**Comment by hailin0** [comment]
LGTM

**Review by hailin0 — APPROVED** [review]

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-googlesheets-e2e/src/test/resources/fakesource_to_googlesheets.conf [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+source {
+  FakeSource {
+    row.num = 3
+    schema = {
+      fields {
+        name = "string"
+        address = "string"
+        age = "int"
```
Comment (TyrantLucifer) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-googlesheets-e2e/src/test/resources/fakesource_to_googlesheets.conf:30]: test all data types

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]
Please offer test screenshots.

**Comment by TyrantLucifer** [comment]
Any process this pull request?

**Pull request closed without merging**